Python/optimizer_bytecodes.c

```diff
@@ -855,6 +855,17 @@ dummy_func(void) {
         }
     }
 
+    op(_CALL_STR_1, (unused, unused, arg -- res)) {
+        if (sym_matches_type(arg, &PyUnicode_Type)) {
+            // e.g. str('foo') or str(foo) where foo is known to be a string
+            PyObject *value = sym_get_const(ctx, arg);
+            res = sym_new_const(ctx, value);
+        }
```

Comment:
This is an optimization that Brandt suggested. When in `str(foo)`, `foo` is known to be a string, we can propagate it to the output. I also added a test for this case.